areas/api.py

```diff
@@ -0,0 +1,89 @@
+from collections import defaultdict
+
+from munigeo.models import AdministrativeDivision
+from parler_rest.fields import TranslatedFieldsField
+from parler_rest.serializers import TranslatableModelSerializer
+from rest_framework import mixins, serializers, viewsets
+
+
+class TranslatedModelSerializer(TranslatableModelSerializer):
+    translations = TranslatedFieldsField()
+
+    def to_representation(self, obj):
+        ret = super(TranslatedModelSerializer, self).to_representation(obj)
+        if obj is None:
+            return ret
+        return self.translated_fields_to_representation(obj, ret)
+
+    def translated_fields_to_representation(self, obj, ret):
+        translated_fields = {}
+
+        for lang_key, trans_dict in ret.pop('translations', {}).items():
+
+            for field_name, translation in trans_dict.items():
+                if field_name not in translated_fields:
+                    translated_fields[field_name] = {lang_key: translation}
+                else:
+                    translated_fields[field_name].update({lang_key: translation})
+
+        ret.update(translated_fields)
+
+        return ret
+
+
+class AdministrativeDivisionSerializer(TranslatedModelSerializer):
+    bbox = serializers.ReadOnlyField(source='geometry.boundary.extent')
+
+    class Meta:
+        model = AdministrativeDivision
+        fields = ('translations', 'ocd_id', 'origin_id', 'bbox')
+
+
+class NeigborhoodSerializer(AdministrativeDivisionSerializer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['sub_districts'] = serializers.SerializerMethodField()
+
+        sub_district_qs = AdministrativeDivision.objects.filter(
+            type__type='sub_district'
+        ).exclude(
+            origin_id='Aluemeri'  # wtf
+        ).select_related(
+            'geometry'
+        ).prefetch_related(
+            'translations'
+        ).order_by(
+            'origin_id'
+        )
+
+        # cache sub districts grouped by their parent origin_id for faster access
+        sub_district_map = defaultdict(list)
+        for sub_district in sub_district_qs:
+            origin_id = sub_district.origin_id
+            if origin_id.endswith('0'):
+                # When a sub district id ends with '0', it means the neighborhood consists of only one sub district,
+                # which for us means basically "the neighborhood has no sub districts", so we skip the sub district.
+                # For example neighborhood Konala 32 has only one sub district, Konala 320.
+                continue
+            parent_origin_id = origin_id[:2]
```

Comment:
aren't there IDs like `10` (kruununhaka) and `20` (kluuvi)? or are those `010` and `020`

Reply:
We skip everything that ends with `0`, so we skip those as well. And it is enough in this case, because neighborhoods with ID < 10 don't have sub districts. But yeah, this isn't really robust as it is now, I used to have some more checks there but lost them somewhere along the way. But let's go with this now to get it online, and improve it soon(tm)

Reply:
I just meant that if `origin_id` is `10` you will get the same `10` for `parent_origin_id` 🙂

Reply:
Yep, but actually you won't, because you won't reach that line 68 because of the above continue. That isn't pretty atm, but I *think* it works :)